locator/token_metadata.cc

```diff
@@ -844,6 +851,66 @@ future<> token_metadata_impl::update_pending_ranges(
     });
 }
 
+future<> token_metadata_impl::update_topology_change_info(dc_rack_fn& get_dc_rack, bool request_read_new) {
+    if (_bootstrap_tokens.empty() && _leaving_endpoints.empty() && _replacing_endpoints.empty()) {
+        co_await utils::clear_gently(_topology_change_info);
+        _topology_change_info.reset();
+        co_return;
+    }
+
+    // target_token_metadata incorporates all the changes from leaving, bootstrapping and replacing
+    auto target_token_metadata = co_await clone_only_token_map(false);
+    {
```

Comment:
Why the scope is needed?

Reply:
To restrict the scope of variables needed only to compute `target_token_metadata`. This is somewhat similar to immediately invoked lambdas.